src/sink/rest.rs

```diff
@@ -351,6 +352,7 @@ impl Sink for Rest {
                                 segments.map(String::from).collect()
                             }),
                         };
+                        let response_request = request.clone();
```

Comment:
what about instead of cloning the whole request we build the request metadata here already into a Value and passing this into the `CodecTaskInMsg::ToEvent`?

Reply:
The problem with building the request metadata into a `Value` and then passing it around would be that we would need define a `'static` lifetime in the struct since we are async here, that means when passing it to `CodecTaskInMsg::ToEvent` we would need to `clone_static()` on it, which I presume would be more expensive than caling `.clone()` on `request`. Am I correct here?

Reply:
My idea here was to create a`Value<'static>` and to achieve this clone request headers, url etc to an owned value (like `String` and then use `Value::from(...)` on that, as all the containing values in that also need to be `'static`. It could be, I am mistaken here. @Licenser do you know if this could fly? Seems to be the most efficient way to extract the needed data from the request and to send it across an async boundary.